src/main/java/foodwhere/logic/parser/RFindCommandParser.java

```diff
@@ -21,15 +23,25 @@ public class RFindCommandParser implements Parser<RFindCommand> {
      * @throws ParseException if the user input does not conform the expected format
      */
     public RFindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, RFindCommand.MESSAGE_USAGE));
+        ArgumentMultimap argMultimap =
+            ArgumentTokenizer.tokenize(args,
+                    CliSyntax.PREFIX_NAME,
+                    CliSyntax.PREFIX_TAG);
+
+        if ((!arePrefixesPresent(argMultimap, CliSyntax.PREFIX_NAME)
+                && !arePrefixesPresent(argMultimap, CliSyntax.PREFIX_TAG))
+            || !argMultimap.getPreamble().isEmpty()) {
```

Comment:
Is this parsing possibly too strict? Perhaps a warning (that the preamble is ignored) can be given if the preamble is nonempty, and the warning suppressed if the preamble only consists of spaces.

Likewise for SFindCommandParser.